cluster_tools/tests/test_all.py

```diff
@@ -35,63 +35,94 @@ def raise_if(msg: str, _bool: bool) -> None:
         raise Exception("raise_if was called with True: {}".format(msg))
 
 
-def get_executors(with_debug_sequential: bool = False) -> List[cluster_tools.Executor]:
-    global _dask_cluster
+# This function is called for each test. It has access to the fixtures supplied
+# to the test and most importantly can parametrize those fixtures.
+# Depending on whether the exc or the exc_with_pickling fixture is used,
+# the test is parametrized with the respective executors.
```

Comment:
fancy :+1: took me a bit to understand this. here's my suggestion for a more high level description.

```suggestion
# Most of the specs in this module should be executed with multiple executors. Some tests
# should be called with all executors (including the pickling variants) and some with a subset (i.e., without the pickling variants).
# In order to avoid redundant parameterization of each test, pytest_generate_tests is defined here.
# If a spec uses an `exc_with_pickling` fixture (defined as a function parameter), that test is automatically parameterized with all executors. Analoguous, parameterization happens with `exc`.
# Regarding how this works in details: This function is called for each test and has access to the fixtures supplied
# to the test and most importantly can parametrize those fixtures.
```